File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function(nums) {
    nums.sort((a,b)=> a-b);
    let l = 0;
    let r = 0;
    let length = 0;

    while (r<nums.length) {
        while (nums[r] - nums[l]>1) {
            l++;
        }
        if (nums[r]-nums[l]===1) {
            length = Math.max(length,r-l+1);
        }
        r++;
    }
    return length;
    
};

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        nums.sort()
        l = r = 0
        length = 0

        while r<len(nums):
            while nums[r]-nums[l]>1:
                l+=1
            if nums[r]-nums[l]==1:
                length = max(length,r-l+1)

            r+=1
        return length